.github/workflows/check_changelog.yml

```diff
@@ -0,0 +1,52 @@
+name: Check Changelog
+
+on:
+  pull_request:
+    paths:
+      - 'pyproject.toml'
+      - 'HISTORY.md'
+
+jobs:
+  check-changelog:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 2
+
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+
+      - name: Install dependencies and check for file changes
+        run: |
+          python -m pip install toml
+          for file in pyproject.toml HISTORY.md; do
+              base_name=$(basename $file .${file##*.})
+              if git diff --name-only HEAD^ | grep -q "^$file$"; then
+                echo "${base_name}_changed=true" >> $GITHUB_ENV
+              else
+                echo "${base_name}_changed=false" >> $GITHUB_ENV
+              fi
+          done
```

Comment:
You don't need to install toml on python 3.12 to read TOML files, you can use https://docs.python.org/3/library/tomllib.html, which is available from 3.11.